### PR TITLE
Don't do format if it's unnecessary

### PR DESCRIPTION
**Pull request by liketic**
**- What I did**
Replace printf or errorf with print, errors.New if the format without arguments. Such as:
```
fmt.Errorf("abc")
```
I think we can use ```errors.New("abc")``` directly.

**- How I did it**

The ```fmt.Errorf ``` calls ```errors.New``` in internal, so if it's unnecessary to format a string, I just replace it with ```errors.New```.
The ```fmt.Fprintf``` if no argument used to format, I just replace it with ```fmt.Fprint```.
The ```t.Fatalf``` if no argument used to format, I just replace it with ```t.Fatal```.

**- How to verify it**
Unit test.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**
👍

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
LGTM, but I'm not sure we should merge right now (https://github.com/docker/docker/pull/28289#issuecomment-259913841)

PTAL @vieux @aaronlehmann

**Comment by ripcurld00d** [comment]
LGTM 2

**Comment by vdemeester** [comment]
I do agree with @AkihiroSuda it might generate unecessary conflicts with potential 1.13 fixes, so I would wait for the release before merging this :angel:

File: builder/dockerfile/bflag.go
Comment (aaronlehmann) [inline builder/dockerfile/bflag.go]: This can pass the string to `panic` instead of an error.

**Comment by thaJeztah** [comment]
@likel sorry, looks like this needs a rebase :cry:

File: daemon/cluster/cluster.go
Comment (thaJeztah) [inline daemon/cluster/cluster.go]: Was discussing this with @vdemeester, and just using `errors.New()` (using the github.com/pkg/errors package) would be ok to do in this file (instead of using the `"errors"` package, and aliasing it as `rerrors`).

Can you change that?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks @likel, I left one comment, but looks good otherwise.

Can you squash your commits? I think for this change it's ok to have a _single_ commit in the pull request

File: daemon/cluster/cluster.go
Comment (likel) [inline daemon/cluster/cluster.go]: Of cource.

**Review by thaJeztah — APPROVED** [review]
thanks! LGTM

**Comment by thaJeztah** [comment]
all green

**Pull request merged**
